### PR TITLE
Add JWT innerSign/encrypt test

### PR DESCRIPTION
**Pull request by sberyozkin**

### CODE REVIEW

File: extensions/smallrye-jwt/deployment/src/test/java/io/quarkus/jwt/test/DefaultGroupsSignEncryptUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    @Test
+    public void echoGroups() {
+        io.restassured.response.Response response = RestAssured.given().auth()
+                .oauth2(token)
+                .get("/endp/echo").andReturn();
+
+        Assertions.assertEquals(HttpURLConnection.HTTP_OK, response.getStatusCode());
+        String replyString = response.body().asString();
+        // The missing 'groups' claim's default value, 'User' is expected
+        Assertions.assertEquals("User", replyString);
```
Comment (gastaldi) [inline extensions/smallrye-jwt/deployment/src/test/java/io/quarkus/jwt/test/DefaultGroupsSignEncryptUnitTest.java:56]: You can use the RestAssured API to perform the assertions: 

```java 
given().auth()
                .oauth2(token)
                .get("/endp/echo")
.then().statusCode(200).
            body(".", equalTo("User"));
```

Comment (sberyozkin) [reply extensions/smallrye-jwt/deployment/src/test/java/io/quarkus/jwt/test/DefaultGroupsSignEncryptUnitTest.java:56]: @gastaldi Sure, it is the copy/paste, I'll update

Comment (sberyozkin) [reply extensions/smallrye-jwt/deployment/src/test/java/io/quarkus/jwt/test/DefaultGroupsSignEncryptUnitTest.java]: Hey @gastaldi - I also ended up cleaning up other tests - many of them have been copied across different projects - some of them require a deeper clean up - but I'll do it the next time I touch some of them...

**Review by gastaldi — APPROVED** [review]
LGTM, good job! 

BTW a minor nitpicking: you can static import `RestAssured.given()` to reduce to only `given()` in your tests ;)

**Comment by sberyozkin** [comment]
@gastaldi Sure, will keep it mind :-)

**Comment by sberyozkin** [comment]
@gastaldi Scala integration tests has failed on Win, 

[The raw logs](https://pipelines.actions.githubusercontent.com/5Na9LjmLgXLiLuzJ0Z1qHRbIDVUXUHNrKe6jPcqGdw5t1S8TiW/_apis/pipelines/1/runs/94445/signedlogcontent/58?urlExpires=2021-01-12T22%3A22%3A18.4574365Z&urlSigningMethod=HMACV1&urlSignature=5WcTrSuxuWoeT9x6qWrCmFxAvRoKulqETYriwCDaTWw%3D) show the PR test as passing so I'll go ahead with the merge

```
2021-01-12T14:39:50.4530139Z [INFO] Running io.quarkus.jwt.test.DefaultGroupsSignEncryptUnitTest
2021-01-12T14:39:52.2122391Z 2021-01-12 14:39:52,069 INFO  [io.quarkus] (main) Quarkus 999-SNAPSHOT on JVM started in 0.755s. Listening on: http://localhost:8081
2021-01-12T14:39:52.2479134Z 2021-01-12 14:39:52,069 INFO  [io.quarkus] (main) Profile test activated. 
2021-01-12T14:39:52.2889239Z 2021-01-12 14:39:52,069 INFO  [io.quarkus] (main) Installed features: [cdi, mutiny, resteasy, resteasy-jsonb, security, smallrye-context-propagation, smallrye-jwt, vertx, vertx-web]
2021-01-12T14:39:53.6169415Z 2021-01-12 14:39:53,482 INFO  [io.quarkus] (main) Quarkus stopped in 0.022s
2021-01-12T14:39:53.8219799Z [INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 3.608 s - in io.quarkus.jwt.test.DefaultGroupsSignEncryptUnitTest
```

**Pull request merged**